src/ui/modules/urlExport.ts

```diff
@@ -21,31 +21,28 @@ const responeHandler = (request: XMLHttpRequest): string => {
   return '🎉 Design tokens pushed to server!'
 }
 
-const urlExport = (parent, exportSettings: urlExportSettings, requestBody: urlExportRequestBody) => {
-  // abort on missing url
-  if (exportSettings.url === '') {
-    // @ts-ignore
-    parent.postMessage({
-      pluginMessage: {
-        command: commands.closePlugin,
-        payload: {
-          notification: '🚨 No server url was provided, push aborted!'
-        }
-      } as PluginMessage
-    }, '*')
-  }
-  // init request
-  const request = new XMLHttpRequest()
-  // send to user defined url
-  request.open('POST', exportSettings.url)
+const ACCEPT_HEADER_KEY = 'Accept'
+const CONTENT_TYPE_HEADER_KEY = 'Content-Type'
+const AUTHORIZATION_HEADER_KEY = 'Authorization'
+
+const addUrlExportRequestHeaders = (request: XMLHttpRequest, exportSettings: urlExportSettings) => {
   // set request header if provided
-  request.setRequestHeader('Accept', exportSettings.acceptHeader || 'application/vnd.github.everest-preview+json')
+  if (exportSettings.authType !== config.key.authType.gitlabToken) {
+    request.setRequestHeader('Accept', exportSettings.acceptHeader || 'application/vnd.github.everest-preview+json')
+  }
+
   // set Content-Type header if provided
-  request.setRequestHeader('Content-Type', exportSettings.contentType || 'text/plain;charset=UTF-8')
-  // add access token of provided
+  if (exportSettings.authType !== config.key.authType.gitlabToken) {
```

Comment:
This could be added into the if clause above, right? As you have the same if clause twice. 

```js
if (exportSettings.authType !== config.key.authType.gitlabToken) {
    request.setRequestHeader('Accept', exportSettings.acceptHeader || 'application/vnd.github.everest-preview+json')
    request.setRequestHeader('Content-Type', exportSettings.contentType || 'text/plain;charset=UTF-8')
  }
```

Or am I missing something here?

Reply:
@lukasoppermann this change is verry similar to what I had originally had. It just didn't include the if statement for the Authorization header. I was under the impression you didn't want this but I can easily update this to wrap Accept and Content-Type

Reply:
I think this was a missunderstanding, sorry.

Nesting ifs inside each other is something I would love to avoid.
```js
if (condition) {
  if (other condition) {
  // ....
  }
}
```

However doing multiple things within the same if clause is totally fine.
Does this make sense?